File: jobs/2026-04-25__23-53-07/hatchet_typescript_worker__S74WXok/artifacts/code/worker.ts
import Hatchet, { Workflow } from '@hatchet-dev/typescript-sdk';
import * as fs from 'fs';
import * as path from 'path';

const hatchet = Hatchet.init();

const workflow: Workflow = {
  id: 'user-create-workflow',
  description: 'Greet the user',
  on: {
    event: 'user:create',
  },
  steps: [
    {
      name: 'greet',
      run: async (ctx) => {
        const input = ctx.workflowInput();
        const username = input.username;
        const result = { message: `Welcome ${username}` };
        
        const logPath = '/home/user/app/output.log';
        fs.writeFileSync(logPath, JSON.stringify(result));
        
        return result;
      },
    },
  ],
};

async function main() {
  const worker = await hatchet.worker('test-worker');
  await worker.registerWorkflow(workflow);
  worker.start();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: jobs/2026-05-26__11-38-52/bulk_run_py__2vgfTvr/artifacts/user/myproject/worker.py
import os
from hatchet_sdk import Hatchet, Context
from pydantic import BaseModel

hatchet = Hatchet()

class InputPayload(BaseModel):
    n: int

@hatchet.task(input_validator=InputPayload)
def double_value(input: InputPayload, context: Context):
    return {"result": input.n * 2}

def main():
    worker = hatchet.worker('bulk-worker')
    worker.register_workflow(double_value)
    worker.start()

if __name__ == "__main__":
    main()
